### PR TITLE
Unit tests for deploying code

### PR DESCRIPTION
**Pull request by johnnyman727**

### CODE REVIEW

File: test/unit/name.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ exports['Tessel.prototype.rename'] = {
   },
 
   tearDown: function(done) {
+    this.tessel.close();
```
Comment (johnnyman727) [inline test/unit/name.js]: This fixed a memory leak where the process listener for SIGINT was never removed.

File: lib/controller.js [hunk 2]
```diff
@@ -283,7 +283,10 @@ function deployScript(opts, push) {
   return Tessel.get(opts)
     .then(function(tessel) {
       // Run the script on Tessel
-      tessel.deployScript(opts, push);
+      return tessel.deployScript(opts, push)
+        .then(function() {
+          tessel.connection.end();
```
Comment (johnnyman727) [inline lib/controller.js]: We want to make sure we always close connections after we're done with them.

**Comment by johnnyman727** [comment]
This is dependent on the USB fixes in #167. Otherwise, it's going to have the same errors when the connection is attempted to be closed.

File: test/unit/name.js (hunk 1 above)
Comment (rwaldron) [inline test/unit/name.js]: Nice!

File: test/common/remote-process-simulator.js [hunk 3]
```diff
@@ -13,10 +14,27 @@ function RemoteProcessSimulator() {
 
   this.control._write = function(command, enc, cb) {
     // Emit commands for validation in tests
-    this.emit('command', command);
+    self.emit('command', command);
     // Call the callback so we can receive more
     cb();
-  }.bind(this);
+  };
+
```
Comment (rwaldron) [inline test/common/remote-process-simulator.js]: This changeset creates a very minor conflict—easily resolved.

File: test/common/remote-process-simulator.js [hunk 4]
```diff
@@ -13,10 +14,27 @@ function RemoteProcessSimulator() {
 
   this.control._write = function(command, enc, cb) {
     // Emit commands for validation in tests
-    this.emit('command', command);
+    self.emit('command', command);
```
Comment (rwaldron) [inline test/common/remote-process-simulator.js]: When correcting the conflict mentioned below, don't forget to update this to `'control'`

Comment (johnnyman727) [inline test/common/remote-process-simulator.js]: Ah good catch! Will fix.

**Comment by johnnyman727** [comment]
@rwaldron thanks! I also made one more fix to delete the `funkytown` folder that existed after tests complete.

**Pull request merged**